modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc

```diff
@@ -0,0 +1,136 @@
+
+[NOTE]
+====
+To provision bare-metal hosts to the cluster by using the BMO, you must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`.
+====
+
+.Prerequisites
+
+* You created a user-provisioned bare-metal cluster.
+* You have baseboard management controller (BMC) access to the hosts.
+* You deployed a provisioning service in the cluster by creating a `Provisioning` CR.
+
+.Procedure
+
+. Create the `Secret` CR and the `BareMetalHost` CR.
+
+.. Save the following YAML in the `bmh.yaml` file:
++
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: worker1-bmc
+  namespace: openshift-machine-api
+type: Opaque
+data:
+  username: <base64_of_uid>
+  password: <base64_of_pwd>
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: worker1
+  namespace: openshift-machine-api
+spec:
+  bmc:
+    address: <protocol>://<bmc_url> <1>
+    credentialsName: "worker1-bmc"
+  bootMACAddress: <nic1_mac_address>
+  externallyProvisioned: false <2>
+  customDeploy:
+    method: install_coreos
+  online: true 
+  userData:
+    name: worker-user-data-managed
+    namespace: openshift-machine-api
+----
+<1> You can only use bare-metal host drivers that support virtual media networking booting, for example `redfish-virtualmedia` and `idrac-virtualmedia`.
+<2> You must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`. The default value is `false`.
+
+. Create the bare-metal host object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f bmh.yaml
+----
++
+.Example output
+[source,terminal]
+----
+secret/worker1-bmc created                    
+baremetalhost.metal3.io/worker1 created
+----
+
+. Approve all certificate signing requests (CSRs).
+
+.. Verify that the provisioning state of the host is `provisioned` by running the following command:
++
+[source,terminal]
+----
+$ oc get bmh -A
+----
++
+.Example output
+[source,terminal]
+----
+NAMESPACE               NAME          STATE                    CONSUMER   ONLINE   ERROR   AGE                      
+openshift-machine-api   controller1   externally provisioned              true             5m25s                    
+openshift-machine-api   worker1       provisioned                         true             4m45s                    
+----
+
+.. Get the list of pending CSRs by running the following command:
++
+[source,terminal]
+----
+$ oc get csr
+----
++
+.Example output
+[source,terminal]
+----
+NAME        AGE   SIGNERNAME                                    REQUESTOR                                         REQUESTEDDURATION CONDITION                                                       
+csr-gfm9f   33s   kubernetes.io/kube-apiserver-client-kubelet   system:serviceaccount:openshift-machine-config-o
```

Comment:
Not sure if this can be fixed but the formatting seems off, I think because of the long requestor field. It may not be possible with our code block formatting to fix but pointing it out just in case.

Reply:
Yeah it looks messed up in the terminal too. I tried fixing as best I could but I just think it's too long. Thanks